packages/sdk-metrics/src/aggregator/exponential-histogram/mapping/LogarithmMapping.ts

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as ieee754 from './ieee754';
+import * as util from '../util';
+import { Mapping, MappingError } from './types';
+
+/**
+ * LogarithmMapping implements exponential mapping functions for scale > 0.
+ * For scales <= 0 the exponent mapping should be used.
+ */
+export class LogarithmMapping implements Mapping {
+  static readonly MIN_SCALE = 1;
+  static readonly MAX_SCALE = 20;
+  private static _PREBUILT_MAPPINGS = new Map<number, LogarithmMapping>();
+
+  /**
+   * get creates or returns a memoized logarithm mapping function for
```

Comment:
Why a memoized function here but premaking all possible mappings in the exponent mapping?

Reply:
This came from the reference implementation, but I don't think this is strong reason to prefer one approach over the other, both have tradeoffs. We could take the memoized function approach with both so that we aren't storing more mappings than we are using. I suspect in a real world application many of the scales will be used, but possibly not all. Should I lazily build the exponent mapping to match this?

Reply:
I dont have a strong preference which we use. I would weakly lean towards premaking all of them instead of memo, but it is up to you. Whatever you decide though I think should be applied to both. Premaking all of the mappings simplifies the implementation a little bit IMO and it isn't such a large number of objects that the memory impact would be noticeable even on small applications.

Reply:
I went ahead and prebuilt the mappings. In doing so, it made sense to consolidate the logic in one place. I put everything in a `getMapping` module, and removed the individual caching / building in logarithm and exponent mapping. This ended up addressing the question about the static `get` methods ([above](https://github.com/open-telemetry/opentelemetry-js/pull/3504#discussion_r1064705874)), as they were removed during this refactor. This will clean things up a bit when I apply them to the accumulation PR.

Reply:
Looks good thanks